test/users-test.js

```diff
@@ -1,10 +1,176 @@
 import chai from 'chai';
 const expect = chai.expect;
-import { bookings, rooms, customers } from './mock-data.js'
 
-describe('See if the tests are running', function() {
-  it('should return true', function() {
-    expect(true).to.equal(true);
+import { bookings, customers, rooms } from './mock-data.js';
+
+import {
+  calcTotalCost,
+  compileUserBookings,
+  createNewBooking,
+  filterRoomsByDate,
+  filterRoomsByType,
+  setCurrentUser,
+  userSelectRoom
+} from '../src/users.js';
+
+describe('Set current user (customer)', () => {
+  it('should find a customer by their id', () => {
+    const currentUser = setCurrentUser(customers, 2);
+
+    expect(currentUser).to.deep.equal({ id: 2, name: 'Rocio Schuster' });
+  });
+
+  it('should find a different customer by their id', () => {
+    const currentUser = setCurrentUser(customers, 4);
+
+    expect(currentUser).to.deep.equal({ id: 4, name: 'Kennedi Emard' });
+  });
+});
+
+describe('Search for available rooms', () => {
+  it('should find rooms available on a specific date', () => {
+    const availableRooms = filterRoomsByDate('2024/01/13', bookings, rooms);
+
+    expect(availableRooms).to.deep.equal([rooms[0], rooms[1], rooms[4]]);
+  });
+
+  it('should find rooms available on a different date', () => {
+    const availableRooms = filterRoomsByDate('2024/01/12', bookings, rooms);
+
+    expect(availableRooms).to.deep.equal([rooms[0], rooms[2], rooms[3], rooms[4]]);
+  });
+
+  it('should fiercly appologize if no rooms are available for a selected date', () => {
+    const allRoomsBooked = [
+      { id: 'book1', userID: 5, date: '2024/01/13', roomNumber: 1 },
+      { id: 'book2', userID: 4, date: '2024/01/13', roomNumber: 2 },
+      { id: 'book3', userID: 3, date: '2024/01/13', roomNumber: 3 },
+      { id: 'book4', userID: 2, date: '2024/01/13', roomNumber: 4 },
+      { id: 'book5', userID: 1, date: '2024/01/13', roomNumber: 5 }
+    ];
+    const availableRooms = filterRoomsByDate('2024/01/13', allRoomsBooked, rooms);
+
+    expect(availableRooms).to.equal(
```

Comment:
If you were to separate out the return message into your DOM updates instead, you could simply check that the return is an empty array.